DatadogCore/Sources/Datadog.swift

```diff
@@ -315,8 +330,8 @@ public struct Datadog {
         trackingConsent: TrackingConsent,
         instanceName: String
     ) throws -> DatadogCoreProtocol {
-        if CoreRegistry.default is DatadogCore {
-            throw ProgrammerError(description: "SDK is already initialized.")
+        guard !CoreRegistry.isRegistered(instanceName: instanceName) else {
+            throw ProgrammerError(description: "The '\(instanceName)' instance of SDK is already initialized.")
         }
```

Comment:
🎯 This is the actual fix for this PR (🐞 solves the problem of instantiating custom instance of the SDK after default one was created).